parsers/fortran.c

```diff
@@ -479,6 +481,15 @@ static tokenInfo *newTokenFrom (tokenInfo *const token)
 	return result;
 }
 
+static tokenInfo *newAnonTokenFrom (tokenInfo *const token, unsigned int uTagKind)
+{
+	tokenInfo *result = newTokenFrom (token);
```

Comment:
Could you add the following change:
```
-static tokenInfo *newTokenFrom (tokenInfo *const token)
+static tokenInfo *newTokenFromFull (tokenInfo *const token, bool noCopyStr)
 {
 	tokenInfo *result = xMalloc (1, tokenInfo);
 	*result = *token;
-	result->string = vStringNewCopy (token->string);
+	result->string = noCopyStr? vStringNewCopy (token->string): vStringNew();
 	token->secondary = NULL;
 	token->parentType = NULL;
 	token->signature = NULL;
 	return result;
 }
 
+static tokenInfo *newTokenFrom (tokenInfo *const token)
+{
+	return newTokenFromFull (token, true);
+}
+
```

So you can use `newTokenFromFull (token, false` here.
As the result you don't have to call `vStringClear (result->string);`.

Reply:
Other than this minor point, all changes look fine.

Reply:
Done.